### ISSUE TITLE
Setting --memory=(pod request memory) is too aggressive 

### ISSUE THREAD

**Issue by travisdowns** [redpanda-data/helm-charts#49]
Currently we set the `--memory` argument of redpanda to use 100.00% of the pod request memory, with no byte left over.

However, this `--memory` parameter doesn't set an absolute cap on the rp process memory use (as seen by cgroups): it is only for the primary memory map for the seastar allocator, but the redpanda process will use a bit than that: for file pages and page tables, at least: these are added the anonymous map size which will eventually grow to almost exactly the `--memory` value.

In practice this causes the rp process to be OOM killed by the kernel as it exceeds its cgroup budget.

So in addition to the larger buffer we need outside the container (for kernel memory, sidecards, etc) we need some buffer _inside_ the container. The page tables are only about 0.2% of the total memory. The file pages seem more variable, but let's say they won't exceed 200 MB.

So we can we use a `--memory` argument which is less than the request amount `R` by `0.002 * R + 200 Mi` ?

**Comment by vuldin** [comment]
From slack:

```
So just to summarize, there are two different "reservations" (or set asides that RP won't use) we are talking about (would be easier on a whiteboard!):
1. The outer one, between the host RAM and the k8s pod limit. E.g., in this demonware case they have 392 GB on the host but the pod limit is 90 GB so they have spare for days (there are other things running there too but nothing very big). This outer buffer has to be fairly large b/c it encompasses all the socket buffers and other kernel memory would could run into the GBs. The 10% value is probably reasonable here (though we should probably try to quantify this more at some point and get to a formula which might look like 1 GB + max_open_connections * sockbuf_size.
2. The inner one, between the container limit (enforced by cgroup) and the redpanda --memory value. We don't need as much buffer here but setting it to 100.00% of the cgroup value doesn't work well either, per above.
```

```
So I think to be comfortable I'd say 200 MB + 0.2%, or if it has to be an absolute number encompassing also very high mem configs at least 1 GB (which is obvs way too big for small containers), since you could have a 400 GB pod without stretching it much and then 0.2% is 800 MB.
```

```
for --memory=200MB, 2MB makes sense… something like that up to 1G of memory. then you do 200MB for 2G of memory, then you do 200MB * .2% for the rest
```

```
I think we could say 200MB + 0.2% * mem since that's almost exactly the same as "200MB + 0.2% * (mem above 2GB)", they differ by only 4 MB and the former is simpler.
```

```
we can use a formula like 200 MB + 0.2% for the in-container reserve.
```

**Comment by vuldin** [comment]
Closed with #145 and #127
